.github/workflows/gitleaks.yml

```diff
@@ -0,0 +1,30 @@
+# Copyright KubeArchive Authors
+# SPDX-License-Identifier: Apache-2.0
+---
+name: gitleaks
+run-name: Running gitleaks check
+# According to the GitHub documentation on Action secrets:
+# "They are not passed to workflows that are triggered by a pull request from a fork."
+# So we can't use pull_requests here since gitleaks needs GITLEAKS_LICENSE.
```

Comment:
This can't run on the pull request, so I have set it to run on the push - when the PR gets merged. This should block a PR that contains a secret. I will create instructions in the CONTRIBUTING document so that devs set up automatic scanning as part of their git configuration. So this is just a second check to catch anything that slips past that.

Reply:
:+1: to ensure that if a credential is accidently checked in, we will get a failed workflow alert.

Something to consider when making this upstream friendly - we can't assume contributors are Red Hat employees. Guiding contributors to add an open source secrets checker to their git configuration would be appropriate.